src/db.c

```diff
@@ -1037,8 +1037,10 @@ void shutdownCommand(client *c) {
             return;
         }
     }
+    blockClient(c, BLOCKED_SHUTDOWN);
```

Comment:
Yeah, it's worth considering. I went for blocking since it's backwards compatible. I imagine there are scripts using `redis-cli shutdown` in a synchronous way (like [this one](https://src.fedoraproject.org/rpms/redis/blob/main/f/redis-shutdown) that was mentioned in issue #9355 recently).

Reply:
ok, let's keep the blocking, but maybe add an INFO field?

thinking out loud again maybe we can also reject write commands?
in normal PAUSE_WRITES we don't wanna reject the commands since we can get un-paused and run them.
if shutdown is terminal, we can refuse them right away.
if we do that, maybe we should also have some new PING response too?

however, IIUC our shutdown is not always terminal (we can abort it), and also, it's expected to be rather quick (just a few seconds), so let's only add the INFO field.